xSDK examples v0.4.0

v-dobrev opened the pull request:
New release, v0.4.0, of the `xsdk-examples` package.

v-dobrev committed:
[xsdk-examples] Initial commit for v0.4.0

v-dobrev committed:
[xsdk-examples] v0.4.0 depends on xsdk@0.8.0

balay committed:
add in missing xsdk dependencies

v-dobrev committed:
[xsdk-examples] remove repeated 'depends_on' directive

alalazo commented on var/spack/repos/builtin/packages/xsdk-examples/package.py:
```diff
@@ -17,6 +17,8 @@ class XsdkExamples(CMakePackage, CudaPackage):
     maintainers("balay", "luszczek", "balos1", "shuds13", "v-dobrev")
 
     version("develop", branch="master")
+    # FIXME: replace the testing branch with the actual release when available:
+    version("0.4.0", branch="testing-v0.4.0")
```
Is this going to happen soon? If so, I'd prefer to wait a couple of days and have the sha256 here.

v-dobrev replied:
The release will happen around mid-May (that's why I added the `WIP` and `don't-merge-yet` labels). The plan is to merge this PR after that. For now, we're just gonna use this branch/PR for testing.

v-dobrev committed:
[xsdk-examples] simplify and extend a bit the package

[mfem] process more optional dependencies of HiOp

[strumpack, superlu-dist] add a workaround for an issue on Mac

v-dobrev committed:
[mfem] fix the handling of the hiop dependency

v-dobrev commented:
@spackbot help

v-dobrev commented:
@spackbot fix style

v-dobrev committed:
[@spackbot] updating style on behalf of v-dobrev

v-dobrev committed:
[xsdk-examples] enable 'heffte' and 'tasmanian' if enabled in 'xsdk'

v-dobrev committed:
[xsdk-examples] Add PUMI dependency

v-dobrev committed:
[xsdk-examples] Add preCICE dependency

balay committed:
[xsdk-examples] add +rocm

balay committed:
heffte: add in a backport fix for building xsdk-examples with cuda

v-dobrev committed:
[xsdk] Remove the explicit requirement for deal.II to be built +hdf5

balay committed:
ENABLE_ROCM -> ENABLE_HIP

v-dobrev committed:
[hiop] Workaround for CMake not finding Cray's BLAS (libsci)

[xsdk-examples] Set CUDA/HIP architectures; sync cuda/rocm variants with xsdk

v-dobrev commented:
@spackbot fix style

v-dobrev committed:
[@spackbot] updating style on behalf of v-dobrev

v-dobrev committed:
[exago] Workaround for CMake not finding Cray's LAPACK/BLAS, libsci

[mfem] Tweaks for running tests under Flux and PBS

v-dobrev committed:
[slate] Pass CUDA/HIP architectures to CMake

v-dobrev committed:
[heffte] For newer CMake versions, set CMAKE_CUDA_ARCHITECTURES

v-dobrev committed:
[hypre] Patch v2.26.0 to fix sequential compilation in 'src/seq_mv'

v-dobrev committed:
[xsdk-examples] Some tweaks in dependencies and compilers used

v-dobrev committed:
[xsdk] Make the 'trilinos' variant sticky

[xsdk-examples] Tweak dependencies

v-dobrev committed:
[slate] Fix copy-paste error

v-dobrev committed:
Merge branch 'develop' into packages/xsdk-examples-v0.4.0

v-dobrev committed:
[xsdk-examples] Workaround for CMakePackage not having the legacy
   property 'build_directory'

v-dobrev commented on var/spack/repos/builtin/packages/xsdk-examples/package.py:
```diff
@@ -103,5 +103,5 @@ def enabled(pkg):
 
     def check(self):
-        with working_dir(self.build_directory):
+        with working_dir(self.builder.build_directory):
```
@alalazo,  FYI:

This looks like a general Spack issue:

After merging the latest `develop` branch in this PR, `self.build_directory` was no longer working with error message
```console
'XsdkExamples' object has no attribute 'build_directory'
```
I had to replace it with `self.builder.build_directory` to get it to work.

mgates3 reviewed (commented):
The SLATE changes look good. I didn't look at other packages.

v-dobrev committed:
[xsdk-examples] Replace the testing branch used temporarily for v0.4.0 with
                the official release

v-dobrev commented:
This branch is now ready for final review and merging.

v-dobrev committed:
Merge branch 'develop' into packages/xsdk-examples-v0.4.0

v-dobrev committed:
Merge branch 'develop' into packages/xsdk-examples-v0.4.0

v-dobrev committed:
Merge branch 'develop' into packages/xsdk-examples-v0.4.0

v-dobrev committed:
Merge branch 'develop' into packages/xsdk-examples-v0.4.0
